include/dxc/DxilContainer/DxilPipelineStateValidation.h

```diff
@@ -668,7 +668,8 @@ DxilPipelineStateValidation::CheckedReaderWriter::IncrementPos(size_t size) {
 template <typename _T> inline bool
 DxilPipelineStateValidation::CheckedReaderWriter::Cast(_T **ppPtr, size_t size) {
   PSV_RETB(CheckBounds(size));
-  *ppPtr = reinterpret_cast<_T*>(Ptr + Offset);
+  if (Mode != RWMode::CalcSize)
+    *ppPtr = reinterpret_cast<_T*>(Ptr + Offset);
```

Comment:
It is undefined behavior to apply arithmetic operations to a null pointer. In CalcSize mode Ptr is null.

Reply:
That makes sense then. I guess the check wasn't for `if(Ptr != nullptr)` so that if there is ever another mode where it can be nullptr, this will catch it and we'll decide if that's intentional?

Reply:
Yea. The pointer should be valid when reading and writing, if that changes in the future we can identify and address. Hopefully by then we'll be able to run ubsan as part of our normal testing flow.